compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -87,13 +88,13 @@ pub fn get_linker<'a>(
                     arg.push(format!("{}\\lib\\{}\\store", root_lib_path.display(), a));
                     cmd.arg(&arg);
                 } else {
-                    warn!("arch is not supported");
+                    sess.emit_warning(errors::UnsupportedArch);
                 }
             } else {
-                warn!("MSVC root path lib location not found");
+                sess.emit_warning(errors::MsvcPathNotFound);
             }
         } else {
-            warn!("link.exe not found");
+            sess.emit_warning(errors::LinkExeNotFound);
```

Comment:
Should these migrations be reversed ?

Reply:
Yes, these are logging statements, not diagnostics.

Reply:
Reverted [here](https://github.com/rust-lang/rust/pull/102612/commits/8d94cda8da8e3350f972d5c7a06dae6dc90b4fdd)